detection-rules/link_cloudflare_service_abuse.yml

```diff
@@ -0,0 +1,74 @@
+name: "Suspicious Links to Cloudflare R2 and Edge Services"
+description: "Detects links to Cloudflare R2 storage buckets, Pages, and Workers domains from unsolicited or previously malicious senders who are not on a trusted sender list or have failed DMARC authentication."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          // Check root domain first 
+          .href_url.domain.root_domain in ("r2.dev", "pages.dev", "workers.dev")
+          // Check the specific pub-{hex}.r2.dev subdomain pattern only for r2.dev
+          or (
+              .href_url.domain.root_domain == "r2.dev"
+              and regex.icontains(.href_url.domain.subdomain, '^pub-[a-f0-9]{32}$')
+          )
```

Comment:
why do we need this additional check when the check on line 9 matches the same root_domain?